armi/physics/neutronics/fissionProductModel/fissionProductModel.py

```diff
@@ -205,60 +198,146 @@ def _getAllFissionProductNames(self):
+            )
+            return
+
+        runLog.info(f"Removing the gaseous fission products from the core.")
+        if not isinstance(gasRemovalFractions, dict):
+            raise TypeError(f"The gas removal fractions input is not a dictionary.")
+
+        # Check that the gas removal fractions supplied are within [0.0, 1.0] and cap them
+        # at these limits otherwise.
+        updatedGasRemovalFractions = {}
+        for b, frac in gasRemovalFractions.items():
+            if frac < 0.0:
+                runLog.warning(
+                    f"The fission gas removal fraction is less than zero for {b}. Setting to zero."
+                )
+                updatedGasRemovalFractions[b] = 0.0
+            elif frac > 1.0:
+                runLog.warning(
+                    f"The fission gas removal fraction is greater than one for {b}. Setting to one."
                 )
+                updatedGasRemovalFractions[b] = 1.0
+            else:
+                updatedGasRemovalFractions[b] = frac
+        gasRemovalFractions.update(updatedGasRemovalFractions)
+
+        def _getGaseousFissionProductNumberDensities(b, lfp):
+            """Look into a single lumped fission product object and pull out the gaseous atom number densities."""
+            numberDensities = {}
+            for nb in lfp.keys():
+                if not lumpedFissionProduct.isGas(nb):
+                    continue
+                yld = lfp[nb]
+                ndens = b.getNumberDensity(lfp.name)
+                numberDensities[nb.name] = ndens * yld
+            return numberDensities
+
+        def _removeFissionGasesExplicitFissionProductModeling(
+            core, gasRemovalFractions
+        ):
+            """
+            This is called when `explicitFissionProducts` are selected as the fission product model.
+
+            Notes
+            -----
+            The input provided has the amount of gas to be removed by each block in the core. The
+            gas that remains in a block is first calculated and then the number density of the
+            gaseous nuclides is multiplied by the fraction that remains and the update number
+            density vector for the block is updated.
+            """
+            for b in core.getBlocks():
+                if b not in gasRemovalFractions:
+                    continue
+                removedFraction = gasRemovalFractions[b]
+                remainingFraction = 1.0 - removedFraction
+                updatedNumberDensities = {}
+                for nuc, val in b.getNumberDensities().items():
+                    nb = nuclideBases.byName[nuc]
+                    updatedNumberDensities[nuc] = (
+                        remainingFraction * val
+                        if lumpedFissionProduct.isGas(nb)
+                        else val
+                    )
+                b.updateNumberDensities(updatedNumberDensities)
```

Comment:
I assume that the gas removal fractions are cumulative, but `_removeFissionGasesExplicitFissionProductModeling` is going to be called multiple times.

So if the gas removal fractions are 0.3, 0.4, 0.5 for three successive state points, you would end up with 0.7 * 0.6 * 0.5 = 0.21 of the initial fission gas instead of 0.5. Am I reading this wrong?